node/src/actors/messages.rs

```diff
@@ -198,14 +202,17 @@ pub struct BuildVtt {
     /// List of `ValueTransferOutput`s
     pub vto: Vec<ValueTransferOutput>,
     /// Fee
-    pub fee: u64,
+    pub fee: Fee,
```

Comment:
This is a breaking change for the node JSON-RPC, running a node on this branch I get the following error when trying to call the `sendValue` method using the old format:

```
{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params: invalid type: integer `10000`, expected string or map."},"id":"1"}
```

Reply:
Oops I was missing `#[serde(deserialize_with = "deserialize_fee_backwards_compatible")]` here.

Reply:
Oops I was missing `#[serde(deserialize_with = "deserialize_fee_backwards_compatible")]` here.

Reply:
I see. In that case, the `Untagged` enum from `deserialize_fee_backwards_compatible` does need to handle `u64` as `Absolute`.

Reply:
It does exactly that, I think:

```rust
    Ok(match Untagged::deserialize(deserializer)? {
        Untagged::Fee(fee) => fee,
        Untagged::Float(float) => Fee::relative_from_float(float),
        Untagged::Integer(integer) => Fee::absolute_from_nanowits(integer),
        Untagged::String(string) => string
            .parse::<u64>()
            .map(Fee::absolute_from_nanowits)
            .or_else(|_| string.parse::<f64>().map(Fee::relative_from_float))
            .map_err(serde::de::Error::custom)?,
    })
```

Reply:
Yes, in another comment I had assumed that the `u64` branch would never be used, but it is used now in the node. However the `Untagged::Float` branch is not useful because the fee was never allowed to be a `f64`, the only options were `u64` in the node and u64 encoded as `String` in the wallet. And in the wallet the kind of fee was supplied externally using the `fee_type` field, so the logic above will always parse the fees as absolute, even though they could be relative.

Reply:
> Yes, in another comment I had assumed that the `u64` branch would never be used, but it is used now in the node. However the `Untagged::Float` branch is not useful because the fee was never allowed to be a `f64`, the only options were `u64` in the node and u64 encoded as `String` in the wallet. And in the wallet the kind of fee was supplied externally using the `fee_type` field, so the logic above will always parse the fees as absolute, even though they could be relative.

Gotcha. The API never accepted floats in this field, so there's no reason to start accepting those now. I will remove that case then. Is that ok?

Reply:
> Gotcha. The API never accepted floats in this field, so there's no reason to start accepting those now. I will remove that case then. Is that ok?

@tmpolaczyk so I did this yesterday but I forgot to push the changes and I noticed just now :sweat:

Reply:
So now this is almost backwards compatible, but it's missing the fee type in the wallet. When trying to create a transaction with a relative fee, which is encoded as `{fee: "1", fee_type: "relative"}`, it gets parsed as an absolute fee of 1 nanowit because 1 is an integer. And the transaction is created but with the wrong fee. I believe the only option to fix this is to add back the `fee_type` field, and handling that case explicitly.

Reply:
Gotcha. I'll see what I can do.

Reply:
Done! Now `fee_type`, if provided, overrides the variant of `Fee`! (There's a test for it)